http/public.go

```diff
@@ -94,3 +104,26 @@ var publicDlHandler = withHashFile(func(w http.ResponseWriter, r *http.Request,
 
 	return rawDirHandler(w, r, d, file)
 })
+
+func authenticateShareRequest(r *http.Request, l *share.Link, salt string) (int, error) {
+	if l.PasswordHash == "" {
+		return 0, nil
+	}
+
+	if r.URL.Query().Get("token") == l.Token {
+		return 0, nil
+	}
+
+	password := r.Header.Get("X-SHARE-PASSWORD")
+	if password == "" {
+		return http.StatusUnauthorized, errors.New("unauthorized")
+	}
+	if err := bcrypt.CompareHashAndPassword([]byte(l.PasswordHash), []byte(password+salt)); err != nil {
+		if errors.Is(err, bcrypt.ErrMismatchedHashAndPassword) {
+			return http.StatusUnauthorized, nil
+		}
+		return http.StatusBadRequest, err
+	}
```

Comment:
Sorry, I don't understand the comment. Do you mean that I need to check for a non-zero statuscode above?

Reply:
When catching the `bcrypt.ErrMismatchedHashAndPassword` error, `nil` are being returned instead of `err`, allowing the request to be processed. That way protected shares can be accessed by providing a wrong password.

Reply:
An error here only indicates that there was an error doing the check, but you are right, I didn't check the returned status code of this func